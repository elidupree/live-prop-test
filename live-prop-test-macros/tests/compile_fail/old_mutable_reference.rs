use live_prop_test::live_prop_test;

#[live_prop_test(postcondition = "old(_input)")]
fn run(_input: &mut i32) {}

fn main() {}
