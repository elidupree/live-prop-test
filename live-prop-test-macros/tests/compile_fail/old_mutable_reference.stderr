error: tried to store `old` value of the argument `_input`, which is an &mut reference rather than an owned value. Did you mean `*_input` or `_input.clone()`?
 --> $DIR/old_mutable_reference.rs:3:34
  |
3 | #[live_prop_test(postcondition = "old(_input)")]
  |                                  ^^^^^^^^^^^^^
